src/olympia/accounts/utils.py

```diff
@@ -71,3 +78,60 @@ def path_with_query(request):
 def camel_case(snake):
     parts = snake.split('_')
     return parts[0] + ''.join(part.capitalize() for part in parts[1:])
+
+
+def process_fxa_event(raw_body, **kwargs):
+    """Parse and process a single firefox account event."""
+    # Try very hard not to error out if there's junk in the queue.
+    log = getLogger('accounts.sqs')
+    event_type = None
+    try:
+        body = json.loads(raw_body)
+        event = json.loads(body['Message'])
+        event_type = event.get('event')
+        uid = event.get('uid')
+        timestamp = datetime.fromtimestamp(event.get('ts', ''))
+        if not (event_type and uid and timestamp):
+            raise ValueError(
+                'Properties event, uuid, and ts must all be non-empty')
+    except (ValueError, KeyError, TypeError), e:
+        log.exception('Invalid account message: %s' % e)
+    else:
+        if event_type == 'primaryEmailChanged':
+            email = event.get('email')
+            if not email:
+                log.error('Email property must be non-empty for "%s" event' %
+                          event_type)
+            else:
+                primary_email_change_event.delay(email, uid, timestamp)
+        else:
+            log.debug('Dropping unknown event type %r', event_type)
+
+
+def process_sqs_queue(queue_url, aws_region, queue_wait_time):
+    log = getLogger('accounts.sqs')
+    log.info('Processing account events from %s', queue_url)
+    try:
+        # Connect to the SQS queue.
+        sqs = boto3.client(
+            'sqs', aws_access_key_id=settings.AWS_ACCESS_KEY_ID,
+            aws_secret_access_key=settings.AWS_SECRET_ACCESS_KEY,
+            region_name=aws_region)
+        # Poll for messages indefinitely.
+        while True:
+            # msg = queue.read(wait_time_seconds=queue_wait_time)
+            response = sqs.receive_message(
+                QueueUrl=queue_url, WaitTimeSeconds=queue_wait_time,
+                MaxNumberOfMessages=10)
```

Comment:
Also, given that we do long-polling here by applying `WaitTimeSeconds` I wonder if `MaxNumberOfMessages` is even used. According to the AWS Docs: `Long polling returns messages as soon any message becomes available.` (http://docs.aws.amazon.com/AWSSimpleQueueService/latest/SQSDeveloperGuide/sqs-long-polling.html) but maybe it's a good idea to provide it anyway. The default is 10 too.

Reply:
https://boto3.readthedocs.io/en/latest/reference/services/sqs.html#SQS.Client.receive_message`  boto3's default is 1.  It is used too - I've observed it consuming 10. (the dev queue has a huge backlog).

Reply:
alrighty, I didn't check the boto3 default. Weird that it's different or I just looked at the wrong documentation. All good here though, that's mostly just a note to myself